### PR TITLE
Update style guide with correct usage of MS SQL

### PR DESCRIPTION
**Pull request by achatterjee-grafana**
Added a bulleted list about correct and incorrect usage.

### CODE REVIEW

File: contribute/style-guides/documentation-style-guide.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ The following sections provide general guidelines on topics specific to Grafana
 - Do not use two spaces after a period. Only add one space after each sentence. Do not add a space at the end of the paragraph.
 - Sentence length should be 25 words or less. If your thought is longer than 25 words, consider breaking up the sentence or changing the format to a list.
 - Paragraphs should be three sentences or fewer. Break up long paragraphs.
+- Always use "MS SQL" when referring to MS SQL Server. Do not use "MSSQL", you may see this incorrect usage in some of our UI.
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-style-guide.md:69]: I recommend putting this in the Word Usage section, not here.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: contribute/style-guides/documentation-style-guide.md (hunk 1 above)
Comment (achatterjee-grafana) [reply contribute/style-guides/documentation-style-guide.md]: Done.

File: contribute/style-guides/documentation-style-guide.md [hunk 2]
```diff
@@ -285,3 +285,6 @@ When referencing the Prometheus data source exporters, always use "node_exporter
 
 **Correct:** node_exporter, windows_exporter
 **Incorrect:** Node Exporter, node exporter, Windows Exporter, Windows exporter, windows exporter.
+
+### MS SQL Server
+Always use "MS SQL" when referring to MS SQL Server application. Do not use "MSSQL", you may see this incorrect usage in the UI.
```
Comment (oddlittlebird) [inline contribute/style-guides/documentation-style-guide.md:290]:
```suggestion
Always use "MS SQL" when referring to MS SQL Server application. Do not use "MSSQL" unless you are specifically referring to a UI element that uses it.

Incorrect UI spellings will be corrected in a later version of Grafana.
```

**Review by oddlittlebird — APPROVED** [review]
Refined the second sentence a bit, otherwise good to go. Thanks for updating the style guide!

**Pull request merged**